conans/client/downloaders/download_cache.py

```diff
@@ -78,6 +79,9 @@ def should_upload_sources(package):
             if not path.endswith(".json"):
                 blob_path = os.path.join(path_backups, path)
                 metadata_path = os.path.join(blob_path + ".json")
+                if not os.path.exists(metadata_path):
+                    ConanOutput().warning(f"Missing metadata file for backup source {blob_path}")
```

Comment:
I am not sure about just a warning. This seems like a violation of the consistency / broken storage of backup sources, shouldn't this be a hard error?

Reply:
Makes sense, changed to raise a hard error